File: CodinGame/Puzzles/Easy/MIME Type/MimeType.ts
const N: number = parseInt(readline());
const Q: number = parseInt(readline());

const UNKNOWN = 'UNKNOWN';
const mimeMap = {};

for (let i = 0; i < N; i++) {
  const [ext, mt] = readline().split(' ');
  mimeMap[ext.toLowerCase()] = mt;
}

for (let i = 0; i < Q; i++) {
  const nameSplit = readline().split('.');

  if (nameSplit.length < 2)
    console.log(UNKNOWN);
  else
    console.log(mimeMap[nameSplit.pop().toLowerCase()] || UNKNOWN)
}

File: CodinGame/Puzzles/Easy/MIME Type/MimeType.py
import sys
import math

UNKNOWN = 'UNKNOWN'
N = int(input())
Q = int(input())

mimeMap = {}

for i in range(N):
  ext, mt = input().split()
  mimeMap[ext.lower()] = mt

for i in range(Q):
  fname = input().split('.')

  if len(fname) < 2:
    print(UNKNOWN)
  else:
    print(mimeMap.get(fname[-1].lower(), UNKNOWN))
